packages/core/src/credentials.ts

```diff
@@ -23,7 +25,9 @@ export class Credentials<
 	 * Sets new credential object
 	 */
 	setData(data: T): void {
-		this.data = this.cipher.encrypt(data);
+		const stringified = this.stringifyAndValidate(data);
+
+		this.data = this.cipher.encrypt(stringified);
```

Comment:
Maybe we can skip the additional stringification altogether?

```suggestion
    a.ok(isObjectLiteral(data));
		this.data = this.cipher.encrypt(data);
```

Reply:
We could do that. However it wouldn't then prevent what happened in #12880 since the credential was an object and it got saved in to the DB, but it contained a property that didn't parse into JSON anymore. Let's do it the simpler way here and add schema and validation to the boundary so we don't accept garbage from the outside

Reply:
no code in our stack should be producing unserializable objects. If we are seeing that, then we have an issue elsewhere, that needs to be fixed there IMO.